test/cutensor/elementwise_binary.jl

```diff
@@ -130,12 +130,16 @@ eltypes = ((Float16, Float16),
         @test hD ≈ permutedims(A, p) .- C
 
         α = rand(eltyD)
-        ctD = LinearAlgebra.axpy!(α, ctA, ctC)
+        ctD = CuTensor(copy(ctC.data), copy(ctC.inds))
```

Comment:
I don't think we need to make an actual copy here, but just assert that the returned value is equal (`===`) to the input object. That also doesn't compare actual memory, but just that the objects are the same.

Reply:
Sorry for the late reply. But the test fails if we don't make a copy first because `axpy!` mutates `ctC` which has to be reused in the next test for `axpby!`. However, I did some refactoring which should make things cleaner and clearer.